### PR TITLE
Manually upload windows Github Release using curl

### PR DESCRIPTION
**Pull request by hcorion**
Requesting review from @AniLeo and @JohnHolmesII

### CODE REVIEW

File: .ci/github-upload-windows.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
+#!/bin/sh -ex
+
+ARTIFACT_DIR="$BUILD_ARTIFACTSTAGINGDIRECTORY"
+generate_post_data()
+{
+    body=`cat GitHubReleaseMessage.txt`
```
Comment (JohnHolmesII) [inline .ci/github-upload-windows.sh:6]: Prefer `$()` to ``, per [SC2006](https://github.com/koalaman/shellcheck/wiki/SC2006)

File: .ci/github-upload-windows.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+upload_file()
+{
+    curl -s \
+        -H "Authorization: token ${RPCS3_TOKEN}" \
+        -H "Accept: application/vnd.github.v3+json" \
+        --data-binary @$2\\$3 \
+        "https://uploads.github.com/repos/$repo_full_name/releases/$1/assets?name=$3"
+}
+for file in "$ARTIFACT_DIR"/*; do
+    name=`basename "$file"`
```
Comment (JohnHolmesII) [inline .ci/github-upload-windows.sh:39]: Prefer `$()` to ``, per [SC2006](https://github.com/koalaman/shellcheck/wiki/SC2006)

File: .ci/github-upload-windows.sh [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+repo_full_name="RPCS3/rpcs3-binaries-win"
+
+curl -s \
+    -H "Authorization: token ${RPCS3_TOKEN}" \
+    -H "Accept: application/vnd.github.v3+json" \
+    --data "$(generate_post_data)" "https://api.github.com/repos/$repo_full_name/releases" >> release.json
+
+id=`grep '"id"' release.json | cut -d ' ' -f4 | head -n1`
```
Comment (JohnHolmesII) [inline .ci/github-upload-windows.sh:26]: Prefer `$()` to ``, per [SC2006](https://github.com/koalaman/shellcheck/wiki/SC2006)

File: .ci/github-upload-windows.sh [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+id=${id%?}
+
+upload_file()
+{
+    curl -s \
+        -H "Authorization: token ${RPCS3_TOKEN}" \
+        -H "Accept: application/vnd.github.v3+json" \
+        -H "Content-Type: application/octet-stream" \
+        --data-binary @$2/$3 \
```
Comment (JohnHolmesII) [inline .ci/github-upload-windows.sh:35]: Quote the vars, mostly just to make sc happy (we are unlikely to add globs or spaces, but still)

File: .ci/github-upload-windows.sh [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+repo_full_name="RPCS3/rpcs3-binaries-win"
+
+curl -s \
+    -H "Authorization: token ${RPCS3_TOKEN}" \
+    -H "Accept: application/vnd.github.v3+json" \
+    --data "$(generate_post_data)" "https://api.github.com/repos/$repo_full_name/releases" >> release.json
+
+id=`grep '"id"' release.json | cut -d ' ' -f4 | head -n1`
+id=${id%?}
```
[the remark this answers is not in the document]
Comment (hcorion) [reply .ci/github-upload-windows.sh:27]: Yeah, just trimming. Not sure what you mean by showing an error too?

File: .ci/github-upload-windows.sh [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+repo_full_name="RPCS3/rpcs3-binaries-win"
+
+curl -s \
+    -H "Authorization: token ${RPCS3_TOKEN}" \
+    -H "Accept: application/vnd.github.v3+json" \
+    --data "$(generate_post_data)" "https://api.github.com/repos/$repo_full_name/releases" >> release.json
+
+id=$(grep '"id"' release.json | cut -d ' ' -f4 | head -n1)
+id=${id%?}
+${id:?}
```
Comment (JohnHolmesII) [inline .ci/github-upload-windows.sh:28]: I think this as-is will try and run the value as a command. Need an echo or something.

Comment (hcorion) [reply .ci/github-upload-windows.sh]: Ah yeah that makes sense, good point.

**Pull request merged**